toolbox/src/base/factory.cpp

```diff
@@ -36,7 +36,10 @@ wbt::Block* wbt::Block::instantiateBlockWithClassName(std::string blockClassName
 #ifdef WBT_USES_ICUB
     else if (blockClassName == wbt::MinimumJerkTrajectoryGenerator::ClassName) {
         block = new wbt::MinimumJerkTrajectoryGenerator();
-    } 
+    }
+    else if (blockClassName == wbt::DiscreteFilter::ClassName) {
```

Comment:
Mmh consideing that the last part of the file is all with defines, and the indentation has to be changes, imo the current status looks better. Compare:
```cpp
    ...
    } else if (blockClassName == wbt::GetLimits::ClassName) {
        block = new wbt::GetLimits();
    } else if (blockClassName == wbt::CentroidalMomentum::ClassName) {
        block = new wbt::CentroidalMomentum();
    }
#ifdef WBT_USES_ICUB
    else if (blockClassName == wbt::MinimumJerkTrajectoryGenerator::ClassName) {
        block = new wbt::MinimumJerkTrajectoryGenerator();
    }
    else if (blockClassName == wbt::DiscreteFilter::ClassName) {
        block = new wbt::DiscreteFilter();
    }
#endif
#ifdef WBT_USES_IPOPT
    else if (blockClassName == wbt::InverseKinematics::ClassName) {
        block = new wbt::InverseKinematics();
    }
#endif
```
with
```cpp
    ...
    } else if (blockClassName == wbt::GetLimits::ClassName) {
        block = new wbt::GetLimits();
    } else if (blockClassName == wbt::CentroidalMomentum::ClassName) {
        block = new wbt::CentroidalMomentum();
    }
#ifdef WBT_USES_ICUB
    else if (blockClassName == wbt::MinimumJerkTrajectoryGenerator::ClassName) {
        block = new wbt::MinimumJerkTrajectoryGenerator();
    } else if (blockClassName == wbt::DiscreteFilter::ClassName) {
        block = new wbt::DiscreteFilter();
    }
#endif
#ifdef WBT_USES_IPOPT
    else if (blockClassName == wbt::InverseKinematics::ClassName) {
        block = new wbt::InverseKinematics();
    }
#endif
```